openapi-adoc/src/main/java/io/micronaut/openapi/adoc/utils/SwaggerUtils.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.openapi.adoc.utils;
+
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.openapi.OpenApiUtils;
+import io.swagger.v3.oas.models.OpenAPI;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import static io.micronaut.openapi.adoc.utils.FileUtils.FILE_SCHEME;
+import static io.micronaut.openapi.adoc.utils.FileUtils.loadFileFromClasspath;
+
+/**
+ * File utilities methods.
+ *
+ * @since 5.2.0
+ */
+@Internal
+public final class SwaggerUtils {
+
+    private SwaggerUtils() {
+    }
+
+    public static OpenAPI readOpenApi(String swaggerFileContent, boolean isJson) {
+
+        ObjectMapper mapper;
+        if (isJson) {
+            mapper = OpenApiUtils.getJsonMapper();
+        } else {
+            mapper = OpenApiUtils.getYamlMapper();
+        }
+
+        try {
+            return mapper.readValue(swaggerFileContent, OpenAPI.class);
+        } catch (JsonProcessingException e) {
+            throw new IllegalArgumentException("Can't parse swagger file", e);
+        }
+    }
+
+    public static OpenAPI readOpenApiFromLocation(String location) {
+
+        var isJson = location.endsWith(".json");
+        var adjustedLocation = location.replaceAll("\\\\", "/").toLowerCase();
```

Comment:
Same thing here with path traversal

Reply:
Hm.. i still don't understand, what problem with my code? Your example is not the same, as my case. My location can be in URI format, can be just file system path, can be pah inside jar or classpath. This is the easiest and fastest way to get rid of problems with different slashes in the path and OS

Reply:
ok since this is build time only it is probably fine